examples/broadcasting_your_own_methods.py

```diff
@@ -0,0 +1,472 @@
+    """
+    slippery_region_bl_corner = (400.0, 0.0)
+    slippery_region_tr_corner = (600.0, 2000.0)
+
+    is_within_bounds_x = (
+        slippery_region_bl_corner[0]
+        <= xy_position[0]
+        <= slippery_region_tr_corner[0]
+    )
+    is_within_bounds_y = (
+        slippery_region_bl_corner[1]
+        < xy_position[1]
+        <= slippery_region_tr_corner[1]
+    )
+    return is_within_bounds_x and is_within_bounds_y
+
+
+# We can just check our function with a few sample points
+for point in [(0, 100), (450, 700), (550, 1500), (601, 500)]:
+    print(f"{point} is in slippery region: {in_slippery_region(point)}")
+
+# %%
+# Determine if each position can be seen
+# --------------------------------------
+# Given our data, we could extract whether each position (for each time-point,
+# and each individual) was inside the slippery region by looping over the
+# values.
+
+data_shape = positions.shape
+in_view = np.zeros(
+    shape=(
+        len(positions["time"]),
+        len(positions["keypoints"]),
+        len(positions["individuals"]),
+    ),
+    dtype=bool,
+)  # We would save one result per time-point, per key-point, per individual
+
+# We have turned off the printing for space purposes,
+# but if you re-include the print statements, you'll get a line-by-line
+# update of the progress through the loop.
+for time_index, time in enumerate(positions["time"].values):
+    # print(f"At time {time}:")
+    for keypoint_index, keypoint in enumerate(positions["keypoints"].values):
+        # print(f"\tAt keypoint {keypoint}")
+        for individual_index, individual in enumerate(
+            positions["individuals"].values
+        ):
+            xy_point = positions.sel(
+                time=time,
+                keypoints=keypoint,
+                individuals=individual,
+            )
+            was_in_view = in_slippery_region(xy_point)
+            was_seen = (
+                "was in slippery region"
+                if was_in_view
+                else "was not in slippery region"
+            )
+            # print(
+            #     "\t\tIndividual "
+            #     f"{positions['individuals'].values[individual_index]} "
+            #     f"{was_seen}"
+            # )
+            # Save our result to our large array
+            in_view[time_index, keypoint_index, individual_index] = was_in_view
+
+# %%
+# We could then build a new ``DataArray`` to store our results, so that we can
+# access the results in the same way that we did our original data.
+was_in_slippery_region = xr.DataArray(
+    in_view,
+    dims=["time", "keypoints", "individuals"],
+    coords={
+        "time": positions["time"],
+        "keypoints": positions["keypoints"],
+        "individuals": positions["individuals"],
+    },
+)
+
+print(
+    "Boolean DataArray indicating if at a given time, "
+    "a given individual was inside the slippery region:"
+)
+print(was_in_slippery_region)
```

Comment:
For printing data arrays (here and in other occurences in this example), I suggest just calling the variable name (instead of printing):
```suggestion
was_in_slippery_region
```

sphinx-gallery renders that as an interactive html widget, allowing viewers to explore the data array interactively.